### ISSUE TITLE
Proposal for change/nerfs to constructor

### ISSUE THREAD

**Issue by Pextri** [Eggs-D-Studios/wos-issues#9]
### Guidelines

- [X] I checked for duplicate suggestions
- [X] My suggestion intends to add or improve something

### Version

Development (Unstable)

### Topic

A design problem (e.g. clunky UX, a balancing issue, etc)

### Why are you making your suggestion?

Constructor is currently able to be used for a few abusable combat strategies, such as magically teleporting destructive devices such as bombs or obscructions/anchor devices ect, long distance or inside buildings/ships (with or without the help of parts like life scanner), you could use this to make base defenses useless. All this along with its ability to stay tucked away within the safety of your ship, spawning bombs and missiles unhindered outside your ship, with no exposure to the outside world. If constructor is able to have effects like this, it should be reworked in a way to give it the same treatment you would expect around any item used for combat

### What do you propose? What should change, or what should be added?

Add a delay for spawning models after the constructor has been triggered, I think something long enough to prevent you from being able to intercept a moving target in most cases, to make constructor offset abuse much less practical

Add power cost for constructing (possibly based on material cost) maybe even allow it to pull materials/energy from bins and powercells to fill whatever powercells and bins are being constructed. This would make construction much more taxing than just requiring materials from a bin, while making stuff such as drone construction much better.

Nerf constructor range (it currently allows you to magically deploy a nuke anywhere like it was a Stargate transporter)

Prevent explosives and other devices from being triggered/having effect for a short bit after spawn, possibly no-clip as well. Had the funny concern of users spawning giant walls/jails around ships

Require constructor to have a direct line of sight to the center area where it's spawning the model/require the model to fit in the space its spawning. This allow for it to be targeted in the case of missile/bomb spawning outside of the build, instead of allowing it to stay in a guarded box deploying objects to any location.

Other ideas: make constructor required for model loading (player required to be in constructor proximity with all materials available) this would limit players to have the same loading limitations as constructor without loading from it directly

**Comment by L4PRY** [comment]
Delaying the spawn of the constructed model is a hard-fix to jailing and exploding ships, but I have to say I'm not too fond of the idea. Understandably, as of now, constructors need to be nerfed.
I believe the delay should be based on distance from the constructor, so it'll still allow for tactical placement of explosives as a counter to missiles. I've been meaning to make this happen, but without a proper way to get the range to an object its hardly possible. In unstable, however, the raycast parts would make this a lot easier, that's if the constructors allow it to.

**Comment by Pextri** [comment]
I agree with some of these, however, I feel the constructor is already quite big in size, and making it larger would just add more unnecessary space to it, rather than addressing the actual functional issues it is capable of currently.
